pkg/workflows/interfaces/interfaces.go

```diff
@@ -92,3 +94,10 @@ type ClusterCreator interface {
 	Run(ctx context.Context, spec *cluster.Spec, managementCluster types.Cluster) error
 	CreateSync(ctx context.Context, spec *cluster.Spec, managementCluster *types.Cluster) (*types.Cluster, error)
 }
+
+// EksaInstaller exposes the EKSA installer methods.
+type EksaInstaller interface {
+	ApplyBundles(ctx context.Context, log logr.Logger, cluster *types.Cluster, spec *cluster.Spec) error
+	ApplyReleases(ctx context.Context, log logr.Logger, cluster *types.Cluster, spec *cluster.Spec) error
+	CreateNamespaceIfNotPresent(ctx context.Context, log logr.Logger, cluster *types.Cluster, spec *cluster.Spec) error
+}
```

Comment:
ideally we should have
```suggestion
type EksaInstaller interface {
	Install(ctx context.Context, log logr.Logger, cluster *types.Cluster, spec *cluster.Spec) error
}
```

if we can't do that for some reason, let's talk about why and try to figure out a solution